auth/src/main/java/org/aerogear/auth/credentials/IntegrityCheckParameters.java

```diff
@@ -0,0 +1,49 @@
+package org.aerogear.auth.credentials;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class IntegrityCheckParameters implements IIntegrityCheckParameters {
+
+    private String issuer;
+    private String audience;
+    private String publicKey;
+
+    public IntegrityCheckParameters(final String audience, final String issuer, final String publicKey) {
+        this.issuer = issuer;
+        this.audience = audience;
+        this.publicKey = publicKey;
+    }
+
+    public IntegrityCheckParameters() {}
+
+    public String getIssuer() {
+        return this.issuer;
+    }
+
+    public String getAudience() {
+        return this.audience;
+    }
+
+    public String getPublicKey() { return this.publicKey; }
+
+    public String serialize() throws JSONException {
+        return new JSONObject()
+            .put("audience", this.audience)
+            .put("issuer", this.issuer)
+            .put("publicKey", this.publicKey)
+            .toString();
+    }
+
+    public static IntegrityCheckParameters deserialize(final String serializedParams) throws JSONException {
```

Comment:
Is this object supposed to be constructed only through this method? If yes, the constructor should be private

Reply:
No this will need to be constructed through the constructor when initially created at authentication time.